### PR TITLE
improvement: Match completions on union types

### PR DESCRIPTION
**Pull request by jkciesluk**
closes https://github.com/scalameta/metals-feature-requests/issues/371

### CODE REVIEW

File: tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala [hunk 1]
```diff
@@ -34,12 +29,7 @@ class CompletionMatchSuite extends BaseCompletionSuite {
       |}""".stripMargin,
     """|match
        |match (exhaustive) Option[Int] (2 cases)
-       |""".stripMargin,
-    compat = Map(
-      "3" -> """|match
-                |match (exhaustive) Option (2 cases)
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala:40]: We might need to adjust the results for nightly Scala version.

File: tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala [hunk 2]
```diff
@@ -853,4 +841,19 @@ class CompletionMatchSuite extends BaseCompletionSuite {
...
+    """
+      |case class Foo(a: Int)
+      |case class Bar(b: Int)
+      |
+      |object O {
+      |  val x: Foo | Bar = ???
+      |  val y  = x match@@
+      |}""".stripMargin,
+    """|match
+       |match (exhaustive) Foo | Bar (2 cases)
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala:855]: Could you also add checkEdit?

File: tests/cross/src/test/scala/tests/pc/CompletionCaseSuite.scala [hunk 3]
```diff
@@ -793,4 +793,20 @@ class CompletionCaseSuite extends BaseCompletionSuite {
...
+      |case class Foo(a: Int)
+      |case class Bar(b: Int)
+      |
+      |object O {
+      |  val x: Foo | Bar = ???
+      |  val y  = List(x).map{ ca@@ }
+      |}""".stripMargin,
+    """|case Bar(b) => union-type
+       |case Foo(a) => union-type
+       |case (exhaustive) Foo | Bar (2 cases)
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/pc/CompletionCaseSuite.scala:808]: Also checkEdit for the exhaustive case?

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala (hunk 1 above)
Comment (jkciesluk) [reply tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala:40]: I've ignored the tests for now, until we backport to Dotty.

**Review by tgodzik — APPROVED** [review]
LGTM

**Pull request merged**